File: app/src/main/java/com/kelsos/mbrc/commands/CancelNotificationCommand.java
package com.kelsos.mbrc.commands;

import com.google.inject.Inject;
import com.kelsos.mbrc.interfaces.ICommand;
import com.kelsos.mbrc.interfaces.IEvent;
import com.kelsos.mbrc.messaging.NotificationService;

public class CancelNotificationCommand implements ICommand {
  private NotificationService notificationService;

  @Inject public CancelNotificationCommand(NotificationService notificationService) {
    this.notificationService = notificationService;
  }

  @Override public void execute(IEvent e) {
    notificationService.cancelNotification(NotificationService.NOW_PLAYING_PLACEHOLDER);
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/commands/CancelNotificationCommand.kt
package com.kelsos.mbrc.commands

import com.kelsos.mbrc.interfaces.ICommand
import com.kelsos.mbrc.interfaces.IEvent
import com.kelsos.mbrc.messaging.NotificationService
import javax.inject.Inject

class CancelNotificationCommand
@Inject constructor(private val notificationService: NotificationService) : ICommand {

  override fun execute(e: IEvent) {
    notificationService.cancelNotification(NotificationService.NOW_PLAYING_PLACEHOLDER)
  }
}
